### DISCUSSION TITLE
get file info from CLI

### DISCUSSION THREAD

**Discussion by Travillion** [mkb79/audible-cli#20] [Ideas]
Love these tools, thank you!

I noticed as I've been organizing my library and using your tools to update things like chapter titles, that some of my files have been updated by Audible since they were originally downloaded. For example, a few titles were downloaded in 64 kbps but are now available in 128 kbps. Files were originally downloaded on PC and at default best setting, so it's not a matter of user error. Currently, my only way to tell if a file has since been updated is to try redownloading all my <128 files and comparing the download size. This doesn't take into account any possible corrections issues by the publisher.

Is it possible to build some functionality into CLI to report the file settings (e.g., bitrate) so I can easily compare my source files with "best" on Audible?

**Comment by mkb79** [comment]
Hi. You can do this writing a plugin using mutagen. 

You need the following steps:

1. Get your entire library (take a look on the download command)
2. For every audiobook in your library check if you have already downloaded it (take a look on the download command)
3. If you already have downloaded it make a API [https://audible.readthedocs.io/en/latest/misc/external_api.html#get-1-0-content-asin-metadata](metadata request) and check the available bitrate
4. Use mutagen packages to check the bitrate from downloaded file
5. If the Bitrates doesn't match print out a message or redownload the file. 

I can help you writing this plugin. 

Best regards

**Reply by Travillion** [reply]
Thank you for your swift reply! I am a novice at programming and have never written a plugin. I assume when you say mutagen you are referencing [this: ](https://mutagen.io/documentation/introduction), is that correct?

All of my files have been downloaded and processed (unlocked, renamed to title, and moved into author based folder structure), so I don't expect any plugin to be able to directly compare my local files with those hosted by Audible, though what you described would be awesome!

I was thinking something like just reporting bitrate with the other data included when the csv file of your library is downloaded through your download command. Would this be difficult to implement? Through other audiobook library management software (e.g. MP3Tag), I can sort my titles by bitrate and compare that to the downloaded csv file.

**Reply by mkb79** [reply]
Hi. Sorry for my late answer. I mean [this](https://github.com/quodlibet/mutagen) when I say mutagen. This let you check your audio files for bitrate and compare them with the API response.

**Comment by Travillion** [comment]
Hello,

Sorry it has taken me so long to respond; I got sidetracked with other
issues. In reviewing the Audible API info, I can find references to
retrieving quality (high, normal, extreme, low), but I cannot find anything
that would provide the bitrate. Am I to infer bitrate from quality
category, or is there another API request I'm not seeing that would provide
the desired information?
